src/main/java/org/jenkinsci/plugins/deployment/HostRecords.java

```diff
@@ -0,0 +1,40 @@
+package org.jenkinsci.plugins.deployment;
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+import hudson.model.InvisibleAction;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class HostRecords extends InvisibleAction implements EnvironmentContributingAction {
```

Comment:
New class is being introduced. It makes sense to add Javadoc (with <code>@since</code>) and License headers